app/src/features/surveys/components/SurveySiteSelectionForm.tsx

```diff
@@ -0,0 +1,126 @@
+import { IEditSurveyRequest } from 'interfaces/useSurveyApi.interface';
+import { useContext, useEffect, useState } from 'react';
+import yup from 'utils/YupSchema';
+
+export interface IStratum {
+  survey_stratum_id?: number;
+  name: string;
+  description: string | null;
+}
+
+export interface ISurveySiteSelectionForm {
+  site_selection: {
+    strategies: string[];
+    stratums: IStratum[];
+  };
+}
+
+export const SurveySiteSelectionInitialValues: ISurveySiteSelectionForm = {
+  site_selection: {
+    strategies: [],
+    stratums: []
+  }
+};
+
+export const SurveySiteSelectionYupSchema = yup.object().shape({
+  site_selection: yup.object().shape({
+    strategies: yup
+      .array()
+      .of(yup.string() /* .required('Must select a valid site selection strategy') */)
+      .when('stratums', (stratums: string[], schema: any) => {
+        return stratums.length > 0
+          ? schema.test(
+              'allowsStratums',
+              'You must include the Stratified site selection strategy in order to add Stratums.',
+              (strategies: string[]) => strategies.includes('Stratified')
+            )
+          : schema;
+      }),
+    stratums: yup.array().of(
+      yup.object({
+        survey_stratum_id: yup.number().optional(),
+        name: yup.string().required('Must provide a name for stratum'),
+        description: yup.string().optional()
+      })
+    )
+    /*
+      // TODO assure that duplicate stratums cannot be created
+      .test('duplicateStratums', 'Stratums must have unique names.', (stratums) => {
+        const entries = (stratums || []).map((stratum) => new String(stratum.name).trim());
+        return new Set(entries).size === stratums?.length;
+      })
+      */
+  })
+});
+
+interface ISurveySiteSelectionFormProps {
+  onChangeStratumEntryVisibility: (isVisible: boolean) => void;
+}
+
+/**
+ * Create/edit survey - Funding section
+ *
+ * @return {*}
+ */
+const SurveySiteSelectionForm = (props: ISurveySiteSelectionFormProps) => {
+  const [showStratumDeleteConfirmModal, setShowStratumDeleteConfirmModal] = useState<boolean>(false);
+
+  const formikProps = useFormikContext<IEditSurveyRequest>();
+  const { values, setFieldValue } = formikProps;
+
+  const codesContext = useContext(CodesContext);
+  assert(codesContext.codesDataLoader.data);
+
+  const siteStrategies = codesContext.codesDataLoader.data.site_selection_strategies.map((code) => {
+    return { label: code.name, value: code.name };
+  });
+
+  const handleConfirmDeleteAllStratums = () => {
+    // Delete all Stratums and hide the Stratums form
+    setFieldValue('site_selection.stratums', []);
+    props.onChangeStratumEntryVisibility(false);
+    setShowStratumDeleteConfirmModal(false);
+  };
+
+  const handleCancelDeleteAllStratums = () => {
+    setShowStratumDeleteConfirmModal(false);
+    setFieldValue('site_selection.strategies', [...values.site_selection.strategies, 'Stratified']);
+  };
+
+  useEffect(() => {
```

Comment:
This effect needs to be tweaked, since it causes the delete modal to appear on first render

Reply:
Fixed.